mailet/standard/src/test/java/org/apache/james/transport/matchers/SenderIsRegexTest.java

```diff
@@ -20,99 +20,84 @@
 
 package org.apache.james.transport.matchers;
 
-import java.io.UnsupportedEncodingException;
-import java.util.Collection;
+import static org.assertj.core.api.Assertions.assertThat;
 
 import javax.mail.MessagingException;
 
-import org.apache.james.transport.matchers.SenderIsRegex;
 import org.apache.mailet.MailAddress;
-import org.apache.mailet.Matcher;
-
-public class SenderIsRegexTest extends AbstractSenderIsTest {
-
-    private String regex = ".*";
-
-    public SenderIsRegexTest(String arg0) throws UnsupportedEncodingException {
-        super(arg0);
-    }
-
-    private void setRegex(String regex) {
-        this.regex = regex;
+import org.apache.mailet.base.test.FakeMail;
+import org.apache.mailet.base.test.FakeMailContext;
+import org.apache.mailet.base.test.FakeMatcherConfig;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+
+public class SenderIsRegexTest {
```

Comment:
I mean, there's two way to match things : either the pattern has to match every characters of the value or it can also match is any subpart of the value match the pattern (like grep does)